Add a Vue Apollo example

cilice opened the pull request:
Closes #141 

I had some freedom the pick some kind of a mocking client for apollo since `vue-apollo` doesn't provide one. 
I also used `screen`, is that alright or should i redo it using the destructing approach used everywhere else?

afontcu commented on src/__tests__/vue-apollo.js:
```diff
@@ -0,0 +1,55 @@
+    'alice+new@example.com',
+  )
+
+  mockClient.setRequestHandler(updateUserMutation, variables =>
```
quick question: Is it needed to define this request handler right here? I was reading the text and felt a bit off to have some "setting up" stuff between the `fireEvent.update` and the `fireEvent.click`

cilice replied:
Oh no, totally not, you can define it right in the beginning, next to the query.
This is also highly specific to the mocking client, usually when you come from React's MockProvider you need to define all queries and mutations beforehand.

afontcu replied:
cool 💯! Then I'd suggest moving it to the beginning of the test, so one can easily spot the "Arrange" section of it :)

cilice commented:
I guess I should also add a note this is only for Apollo v2 and will not work with Apollo v3.

afontcu commented on src/__tests__/vue-apollo.js:
```diff
@@ -0,0 +1,62 @@
+
+  await fireEvent.click(screen.getByRole('button', {name: 'Change email'}))
+
+  await waitFor(() => screen.getByText('alice+new@example.com'))
+  expect(screen.getByText('alice+new@example.com')).toBeInTheDocument()
```
question! Can this be simplified to

```suggestion
  await screen.findByText('alice+new@example.com')
```

would it work? Looks like both lines are asserting the same thing (that the text exists)

cilice replied:
It does! I've changed it to `expect(await screen.findByText('Email: alice@example.com')).toBeInTheDocument()`

cilice commented:
Things I've changed now:
- clarified the setup 
- added some remarks about `apollo-client` v2
- added a loading example and added some comments around the needed `await` for data to appear, since fetching is async

I think this PR is in a good state now, feel free to merge. 🙌 ✨

afontcu commented:
@cilice I think this comment went out the minute you pushed, you might have missed it xD

https://github.com/testing-library/vue-testing-library/pull/143#discussion_r429185365

cilice commented:
Oh thats neat, it somehow never crossed my mind to solve it like this. It's great! I've changed it now and squashed it all into one commit.

cilice commented:
@afontcu I think it’s ready to merge 🙏

afontcu commented:
:tada: This PR is included in version 5.0.4 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@testing-library/vue/v/5.0.4)
- [GitHub release](https://github.com/testing-library/vue-testing-library/releases/tag/v5.0.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
